### ISSUE TITLE
Wrong behaviour of .include() .exclude()  when including or excluding @rid 

### ISSUE THREAD

**Issue by temilmg** [orientechnologies/orientdb#6973]
## OrientDB Version, operating system, or hardware.

- v2.2.13

## Operating System
- [ ] Linux

## Expected behavior and actual behavior

select out('fold').include('name') as list from Data where @rid = #38:0
When not including @rid then the name is returned as expected.

select out('fold').include('name','@rid') as list from Data where @rid = #38:0
When adding @rid to the include method then only rids are returned, name is not included. I would expect name to be included as well, anything else is confusing!

Im looking to get a list of names and  @rids returned in one record so no expand(). Is there a workaround?

**Comment by luigidellaquila** [comment]
Hi @temilmg 

The problem here is that, when a field is a document with a valid rid  (positive cluster id and position), it is considered as a normal document so only the rid is sent.
Unfortunately in v.2.2 there is no way to distinguish between a "partial" document, generated with an `.include()` and an actual document, so it's unlikely to be fixed in current release.

In v 3.0 this problem is already addressed: projections will be treated as different types (not ODocument anymore)

As a work-around, you can send to the client the explicit projections with a subquery, eg.
```
select name, @rid as theRid from (
  select expand(out('fold')) from Data where @rid = #38:0
)
```

or even better:

```
select name, @rid as theRid from (
  select expand(out('fold')) from #38:0
)
```

I hope it helps

Thanks

Luigi

**Comment by temilmg** [comment]
Hi Luigi, 
I dont think your workaround is a workaround for my case. Im using nonblocking queries and a remote connection so Im trying to get back a single record from the db server (so its not so chatty). Your workaround gives as many records as there are edges.
The closest I can come to a workaround is to use query below and match the rids with names afterwards.

select out('fold').include("name","otherprop") as props ,out('fold').@rid as rids from #38:0

Im hoping the list order is always the same, otherwise I will end up matching the name with the wrong ids!?

Thank you for the clarification, looking forward to 3.0 :-)

**Comment by luigidellaquila** [comment]
Hi @temilmg 

you are not 100% guaranteed to have the same order, especially under heavy concurrency.
What you can do is following (that should also be more efficient)
```
select out.include("name","otherprop") as props ,out.@rid as rids from (
   select out('fold') as out from #38:0
)
```

Thanks

Luigi

**Comment by temilmg** [comment]
Ok, thank you very much, I will use that instead!

**Comment by lvca** [comment]
Can we close this issue?

**Comment by temilmg** [comment]
I think it could be closed if 3.0 already fixes the include/exclude issue.
Thank you
